### PR TITLE
Create ServerSync component, break out useOfflineStatus into hook

### PR DESCRIPTION
**Pull request by gambinish**
Integrate slimmed down version of https://github.com/NMFS-RADFish/boilerplate/pull/72 into `boilerplate`

https://github.com/NMFS-RADFish/boilerplate/assets/35090461/a80ef72e-d63b-4d0a-bcdc-0ccdc542b69b

Toast seems to be not working as expected. I will open up a separate PR with a few improvements as well.

### CODE REVIEW

File: src/components/ServerSync.jsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const initializeLaunchSequence = async () => {
+    console.log("checking if all data is present...");
+    const homebaseData = await findOfflineData("homebaseData");
+
+    if (!homebaseData.length) {
+      setSyncStatus({ status: false, message: "data not synced, try resyncing" });
+    }
+    // 7 is the amount of data expected from homebase response, but this can be any check
+    if (homebaseData.length === 7) {
+      setSyncStatus({ status: true, message: "all data cached, ready to launch!" });
```
Comment (brockinit) [inline src/components/ServerSync.jsx:51]: Can you make these messages have uppercase first letters?  `All data cached, ready to launch!`

File: src/components/ServerSync.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      localStorage.setItem("lastHomebaseSync", Date.now());
+
+      initializeLaunchSequence();
+      setIsLoading(false);
+    } else {
+      console.log("No network conection, unable to sync with server");
+      console.log("Application in offline mode, using cache.");
+      console.log(
+        `${localStorage.getItem("lastHomebaseSync") ? `Last sync at: ${localStorage.getItem("lastHomebaseSync")}` : "Application has not yet been synced with homebase"}`,
```
Comment (brockinit) [inline src/components/ServerSync.jsx:36]: Please put strings into constants

Comment (brockinit) [reply src/components/ServerSync.jsx:36]: So that the same string isn't copied/pasted in multiple places.

File: src/components/ServerSync.jsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const { isOffline } = useOfflineStatus();
+  const { updateOfflineData, findOfflineData } = useOfflineStorage();
+  const [isLoading, setIsLoading] = useState(false);
+  const [syncStatus, setSyncStatus] = useState("");
+
+  const syncToHomebase = async () => {
+    if (!isOffline) {
+      setIsLoading(true);
+
+      console.log("syncing...");
```
Comment (brockinit) [inline src/components/ServerSync.jsx:20]: Do we need these log statements?

Comment (gambinish) [reply src/components/ServerSync.jsx:20]: I suppose not, I can remove them

**Pull request merged**